### PR TITLE
[BRE-365] - Update Apple Store Creds to Team Issuer Credential

### PR DESCRIPTION
**Pull request by Eeebru**
## Type of change
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [x] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

- Update Apple Store Creds to Team Issuer Credential

## Code changes

* **.github/workflows/build.yml:**  Update to Team Issuer

## Before you submit
- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -701,12 +701,15 @@ jobs:
...
           || github.ref == 'refs/heads/hotfix-rc'
         env:
-          APPLE_ID_USERNAME: ${{ secrets.APPLE_ID_USERNAME }}
-          APPLE_ID_PASSWORD: ${{ secrets.APPLE_ID_PASSWORD }}
-        run: |
-          xcrun altool --upload-app --type ios --file "./bitwarden-export/Bitwarden.ipa" \
-              --username "$APPLE_ID_USERNAME" --password "$APPLE_ID_PASSWORD"
-
+          APP_STORE_CONNECT_TEAM_ISSUER: ${{ secrets.APP_STORE_CONNECT_TEAM_ISSUER }}
+          APP_STORE_CONNECT_AUTH_KEY: 6TV9MKN3GP
```
Comment (michalchecinski) [inline .github/workflows/build.yml:705]: ❗Should the auth key be here in plain text? Isn't that a confidential value?

**Review by michalchecinski — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (Eeebru) [reply .github/workflows/build.yml:705]: That's how we are using it in `clients` repo. I don't know why @vgrassia made it a plain text

Comment (vvolkgang) [reply .github/workflows/build.yml:705]: That's the key identifier and not the actual .p8 key content. This PR is missing setting up that [file](https://github.com/bitwarden/clients/blob/8d507c04a03fec69ed4ca23ae68669bcb0da096b/.github/workflows/build-desktop.yml#L786-L791). 

Any reason why we're not using the [key](https://github.com/bitwarden/ios/blob/main/.github/workflows/build.yml#L366) used in the iOS repo?

Comment (Eeebru) [reply .github/workflows/build.yml:705]: I'm updating that in a minute @vvolkgang, I used the Desktop key unkowing it's only for Desktop

Comment (michalchecinski) [reply .github/workflows/build.yml]: Got it. Thanks for the explanation. ☺️

Comment (vgrassia) [reply .github/workflows/build.yml]: @vvolkgang We use a separate key per repository for security.

**Review by michalchecinski — APPROVED** [review]

**Pull request merged**